src/main/java/org/prebid/server/bidder/conversant/ConversantBidder.java

```diff
@@ -1,30 +1,232 @@
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.BidderUtil;
 import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
 import org.prebid.server.bidder.model.HttpCall;
 import org.prebid.server.bidder.model.HttpRequest;
 import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.conversant.ExtImpConversant;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
 
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 /**
  * Conversant {@link Bidder} implementation.
  */
-public class ConversantBidder implements Bidder {
+public class ConversantBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpConversant>> CONVERSANT_EXT_TYPE_REFERENCE = new
+            TypeReference<ExtPrebid<?, ExtImpConversant>>() {
+            };
+
+    // List of API frameworks supported by the publisher
+    private static final Set<Integer> APIS = IntStream.range(1, 7).boxed().collect(Collectors.toSet());
+
+    // Options for the various bid response protocols that could be supported by an exchange
+    private static final Set<Integer> PROTOCOLS = IntStream.range(1, 11).boxed().collect(Collectors.toSet());
+
+    // Position of the ad as a relative measure of visibility or prominence
+    private static final Set<Integer> AD_POSITIONS = IntStream.range(0, 8).boxed().collect(Collectors.toSet());
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+    private static final String DISPLAY_MANAGER = "prebid-s2s";
+    private static final String DISPLAY_MANAGER_VER = "1.0.1";
+
+    private final String endpointUrl;
+
+    public ConversantBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+    }
 
     @Override
-    public Result<List<HttpRequest>> makeHttpRequests(BidRequest bidRequest) {
-        throw new UnsupportedOperationException();
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        if (bidRequest.getApp() != null) {
+            return Result.emptyWithError(BidderError.badInput("Conversant doesn't support App requests"));
+        }
+
+        final List<BidderError> errors = new ArrayList<>();
```

Comment:
Please extract all logics for creating BidRequest into separate method, like `createBidRequest(...)`